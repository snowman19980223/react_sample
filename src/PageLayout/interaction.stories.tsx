import {Meta} from '@storybook/react'
import {StickyPane, CustomStickyHeader} from './PageLayout.stories'
import {within} from '@storybook/testing-library'
import {expect} from '@storybook/jest'

const meta: Meta = {
  title: 'Layout/PageLayout/interactions',
  // component: PageLayout,
  parameters: {
    layout: 'fullscreen',
    controls: {expanded: true}
  },
  argTypes: {
    // Debug controls
    'Render header?': {
      type: 'boolean',
      defaultValue: true,
      table: {category: 'Debug'}
    },
    'Render pane?': {
      type: 'boolean',
      defaultValue: true,
      table: {category: 'Debug'}
    },
    'Render footer?': {
      type: 'boolean',
      defaultValue: true,
      table: {category: 'Debug'}
    },
    'Header placeholder height': {
      type: 'number',
      defaultValue: 64,
      table: {category: 'Debug'}
    },
    'Pane placeholder height': {
      type: 'number',
      defaultValue: 200,
      table: {category: 'Debug'}
    },
    'Content placeholder height': {
      type: 'number',
      defaultValue: 400,
      table: {category: 'Debug'}
    },
    'Footer placeholder height': {
      type: 'number',
      defaultValue: 64,
      table: {category: 'Debug'}
    },

    // PageLayout prop controls
    containerWidth: {
      type: {
        name: 'enum',
        value: ['full', 'medium', 'large', 'xlarge']
      },
      defaultValue: 'xlarge',
      control: {type: 'radio'},
      table: {category: 'PageLayout props'}
    },
    padding: {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal']
      },
      defaultValue: 'normal',
      control: {type: 'radio'},
      table: {category: 'PageLayout props'}
    },
    rowGap: {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal']
      },
      defaultValue: 'normal',
      control: {type: 'radio'},
      table: {category: 'PageLayout props'}
    },
    columnGap: {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal']
      },
      defaultValue: 'normal',
      control: {type: 'radio'},
      table: {category: 'PageLayout props'}
    },

    // Header prop controls
    'Header.divider.regular': {
      type: {
        name: 'enum',
        value: ['none', 'line']
      },
      defaultValue: 'none',
      control: {
        type: 'radio'
      },
      table: {
        category: 'Header props'
      }
    },
    'Header.divider.narrow': {
      type: {
        name: 'enum',
        value: ['none', 'line', 'filled']
      },
      defaultValue: 'none',
      control: {
        type: 'radio'
      },
      table: {
        category: 'Header props'
      }
    },
    'Header.divider.wide': {
      type: {
        name: 'enum',
        value: ['none', 'line']
      },
      defaultValue: 'none',
      control: {
        type: 'radio'
      },
      table: {
        category: 'Header props'
      }
    },
    'Header.padding': {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal']
      },
      defaultValue: 'none',
      control: {type: 'radio'},
      table: {category: 'Header props'}
    },
    'Header.hidden.regular': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Header props'}
    },
    'Header.hidden.narrow': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Header props'}
    },
    'Header.hidden.wide': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Header props'}
    },

    // Content prop controls
    'Content.width': {
      type: {
        name: 'enum',
        value: ['full', 'medium', 'large', 'xlarge']
      },
      defaultValue: 'full',
      control: {type: 'radio'},
      table: {category: 'Content props'}
    },
    'Content.padding': {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal']
      },
      defaultValue: 'none',
      control: {type: 'radio'},
      table: {category: 'Content props'}
    },
    'Content.hidden.regular': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Content props'}
    },
    'Content.hidden.narrow': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Content props'}
    },
    'Content.hidden.wide': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Content props'}
    },

    // Pane prop controls
    'Pane.position.regular': {
      type: {
        name: 'enum',
        value: ['start', 'end']
      },
      defaultValue: 'end',
      control: {type: 'radio'},
      table: {category: 'Pane props'}
    },
    'Pane.position.narrow': {
      type: {
        name: 'enum',
        value: ['start', 'end']
      },
      defaultValue: 'end',
      control: {type: 'radio'},
      table: {category: 'Pane props'}
    },
    'Pane.position.wide': {
      type: {
        name: 'enum',
        value: ['start', 'end']
      },
      defaultValue: 'end',
      control: {type: 'radio'},
      table: {category: 'Pane props'}
    },
    'Pane.width': {
      type: {
        name: 'enum',
        value: ['small', 'medium', 'large']
      },
      defaultValue: 'medium',
      control: {type: 'radio'},
      table: {category: 'Pane props'}
    },
    'Pane.sticky': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Pane props'}
    },
    'Pane.padding': {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal']
      },
      defaultValue: 'none',
      control: {type: 'radio'},
      table: {category: 'Pane props'}
    },
    'Pane.divider.regular': {
      type: {
        name: 'enum',
        value: ['none', 'line']
      },
      defaultValue: 'none',
      control: {type: 'radio'},
      table: {category: 'Pane props'}
    },
    'Pane.divider.narrow': {
      type: {
        name: 'enum',
        value: ['none', 'line', 'filled']
      },
      defaultValue: 'none',
      control: {type: 'radio'},
      table: {category: 'Pane props'}
    },
    'Pane.divider.wide': {
      type: {
        name: 'enum',
        value: ['none', 'line']
      },
      defaultValue: 'none',
      control: {type: 'radio'},
      table: {category: 'Pane props'}
    },

    // Footer prop controls
    'Footer.divider.regular': {
      type: {
        name: 'enum',
        value: ['none', 'line']
      },
      defaultValue: 'none',
      control: {
        type: 'radio'
      },
      table: {
        category: 'Footer props'
      }
    },
    'Footer.divider.narrow': {
      type: {
        name: 'enum',
        value: ['none', 'line', 'filled']
      },
      defaultValue: 'none',
      control: {
        type: 'radio'
      },
      table: {
        category: 'Footer props'
      }
    },
    'Footer.divider.wide': {
      type: {
        name: 'enum',
        value: ['none', 'line']
      },
      defaultValue: 'none',
      control: {
        type: 'radio'
      },
      table: {
        category: 'Footer props'
      }
    },
    'Footer.padding': {
      type: {
        name: 'enum',
        value: ['none', 'condensed', 'normal']
      },
      defaultValue: 'none',
      control: {type: 'radio'},
      table: {category: 'Footer props'}
    },
    'Footer.hidden.regular': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Footer props'}
    },
    'Footer.hidden.narrow': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Footer props'}
    },
    'Footer.hidden.wide': {
      type: 'boolean',
      defaultValue: false,
      table: {category: 'Footer props'}
    }
  }
}

StickyPane.argTypes = {
  sticky: {
    type: 'boolean',
    defaultValue: true
  },
  numParagraphsInPane: {
    type: 'number',
    defaultValue: 3
  },
  numParagraphsInContent: {
    type: 'number',
    defaultValue: 30
  }
}

const isInViewPort = (boundingRect: {top: number; left: number; right: number; bottom: number}) => {
  return (
    boundingRect.top >= 0 &&
    boundingRect.left >= 0 &&
    boundingRect.right <= document.documentElement.clientWidth &&
    boundingRect.bottom <= document.documentElement.clientHeight
  )
}

StickyPane.play = async ({canvasElement}: {canvasElement: HTMLElement}) => {
  const canvas = within(canvasElement)
  const content3 = await canvas.getByTestId('content3')
  content3.scrollIntoView()
  const paragraph0 = await canvas.getByTestId('paragraph0')
  const paragraphRect = paragraph0.getBoundingClientRect()
  expect(isInViewPort(paragraphRect)).toBe(true)
}

const NonStickyPane = StickyPane.bind({})
NonStickyPane.argTypes = {
  sticky: {
    type: 'boolean',
    defaultValue: false
  },
  numParagraphsInPane: {
    type: 'number',
    defaultValue: 6
  },
  numParagraphsInContent: {
    type: 'number',
    defaultValue: 30
  }
}

NonStickyPane.play = async ({canvasElement}: {canvasElement: HTMLElement}) => {
  const canvas = within(canvasElement)
  const content3 = await canvas.getByTestId('content3')
  content3.scrollIntoView()
  const paragraph0 = await canvas.getByTestId('paragraph0')
  const paragraphRect = paragraph0.getBoundingClientRect()
  expect(isInViewPort(paragraphRect)).toBe(false)
}

CustomStickyHeader.argTypes = {
  sticky: {
    type: 'boolean',
    defaultValue: true
  },
  stickyTop: {
    type: 'string',
    defaultValue: '8rem'
  },
  numParagraphsInPane: {
    type: 'number',
    defaultValue: 10
  },
  numParagraphsInContent: {
    type: 'number',
    defaultValue: 30
  }
}

CustomStickyHeader.play = async ({canvasElement}: {canvasElement: HTMLElement}) => {
  const canvas = within(canvasElement)
  const contentToScroll = await canvas.getByTestId('content3')
  contentToScroll.scrollIntoView()

  const stickyPaneFirstParagraph = await canvas.getByTestId('paragraph0')
  const paragraphBoundaries = stickyPaneFirstParagraph.getBoundingClientRect()
  const stickyHeader = await canvas.getByTestId('sticky-header')
  const stickyHeaderBoundaries = stickyHeader.getBoundingClientRect()
  expect(isInViewPort(paragraphBoundaries)).toBe(true)
  expect(isInViewPort(stickyHeaderBoundaries)).toBe(true)
}
export default meta
export {StickyPane, NonStickyPane, CustomStickyHeader}
