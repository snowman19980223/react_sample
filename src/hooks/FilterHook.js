import {useContext} from 'react'
import {MenuContext} from '../SelectMenuContext'

function useFilter() {
  const {filterText, setFilterText} = useContext(MenuContext)

  const onChange = ev => {
    setFilterText(ev.target.value)
  }

  return [filterText, onChange]
}

export default useFilter
