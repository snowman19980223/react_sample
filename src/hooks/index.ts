export {useOnOutsideClick} from './useOnOutsideClick'
export type {UseOnOutsideClickParameters} from './useOnOutsideClick'
export {useProvidedRefOrCreate} from './useProvidedRefOrCreate'