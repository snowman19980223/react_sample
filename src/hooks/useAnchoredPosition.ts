import React from 'react'
import {PositionSettings, getAnchoredPosition} from '../behaviors/anchoredPosition'
import {useProvidedRefOrCreate} from './useProvidedRefOrCreate'

export interface AnchoredPositionHookSettings extends Partial<PositionSettings> {
  floatingElementRef?: React.RefObject<Element>
  anchorElementRef?: React.RefObject<Element>
}

/**
 * Calculates the top and left values for an absolutely-positioned floating element
 * to be anchored to some anchor element. Returns refs for the floating element
 * and the anchor element, along with the position.
 * @param settings Settings for calculating the anchored position.
 * @param dependencies Dependencies to determine when to re-calculate the position.
 * @returns An object of {top: number, left: number} to absolutely-position the
 * floating element.
 */
export function useAnchoredPosition(
  settings?: AnchoredPositionHookSettings,
  dependencies?: React.DependencyList
): {
  floatingElementRef: React.RefObject<Element>
  anchorElementRef: React.RefObject<Element>
  position: {top: number; left: number} | undefined
} {
  const floatingElementRef = useProvidedRefOrCreate(settings?.floatingElementRef)
  const anchorElementRef = useProvidedRefOrCreate(settings?.anchorElementRef)
  const [position, setPosition] = React.useState<{top: number; left: number} | undefined>(undefined)
  React.useEffect(() => {
    if (floatingElementRef.current instanceof Element && anchorElementRef.current instanceof Element) {
      setPosition(getAnchoredPosition(floatingElementRef.current, anchorElementRef.current, settings))
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, dependencies)
  return {
    floatingElementRef,
    anchorElementRef,
    position
  }
}
