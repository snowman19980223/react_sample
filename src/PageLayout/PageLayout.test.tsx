import React from 'react'
import {act, render} from '@testing-library/react'
import MatchMediaMock from 'jest-matchmedia-mock'
import {ThemeProvider} from '..'
import {viewportRanges} from '../hooks/useResponsiveValue'
import {PageLayout} from './PageLayout'

let matchMedia: MatchMediaMock

describe('PageLayout', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock()
  })

  afterEach(() => {
    matchMedia.clear()
  })

  it('renders default layout', () => {
    const {container} = render(
      <ThemeProvider>
        <PageLayout>
          <PageLayout.Header>Header</PageLayout.Header>
          <PageLayout.Content>Content</PageLayout.Content>
          <PageLayout.Pane>Pane</PageLayout.Pane>
          <PageLayout.Footer>Footer</PageLayout.Footer>
        </PageLayout>
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders condensed layout', () => {
    const {container} = render(
      <ThemeProvider>
        <PageLayout padding="condensed" rowGap="condensed" columnGap="condensed">
          <PageLayout.Header>Header</PageLayout.Header>
          <PageLayout.Content>Content</PageLayout.Content>
          <PageLayout.Pane>Pane</PageLayout.Pane>
          <PageLayout.Footer>Footer</PageLayout.Footer>
        </PageLayout>
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders with dividers', () => {
    const {container} = render(
      <ThemeProvider>
        <PageLayout>
          <PageLayout.Header divider="line" dividerWhenNarrow="filled">
            Header
          </PageLayout.Header>
          <PageLayout.Content>Content</PageLayout.Content>
          <PageLayout.Pane position="start" divider="line" dividerWhenNarrow="filled">
            Pane
          </PageLayout.Pane>
          <PageLayout.Footer dividerWhenNarrow="line">Footer</PageLayout.Footer>
        </PageLayout>
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders pane in different position when narrow', () => {
    const {container} = render(
      <ThemeProvider>
        <PageLayout>
          <PageLayout.Header>Header</PageLayout.Header>
          <PageLayout.Content>Content</PageLayout.Content>
          <PageLayout.Pane positionWhenNarrow="start">Pane</PageLayout.Pane>
          <PageLayout.Footer>Footer</PageLayout.Footer>
        </PageLayout>
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('can hide pane when narrow', () => {
    // Set narrow viewport
    act(() => {
      matchMedia.useMediaQuery(viewportRanges.narrow)
    })

    const {getByText} = render(
      <ThemeProvider>
        <PageLayout>
          <PageLayout.Header>Header</PageLayout.Header>
          <PageLayout.Content>Content</PageLayout.Content>
          <PageLayout.Pane hidden={{narrow: true}}>Pane</PageLayout.Pane>
          <PageLayout.Footer>Footer</PageLayout.Footer>
        </PageLayout>
      </ThemeProvider>
    )

    expect(getByText('Pane')).not.toBeVisible()
  })

  it('shows all subcomponents by default', () => {
    // Set regular viewport
    act(() => {
      matchMedia.useMediaQuery(viewportRanges.regular)
    })

    const {getByText} = render(
      <ThemeProvider>
        <PageLayout>
          <PageLayout.Header>Header</PageLayout.Header>
          <PageLayout.Content>Content</PageLayout.Content>
          <PageLayout.Pane hidden={{narrow: true}}>Pane</PageLayout.Pane>
          <PageLayout.Footer>Footer</PageLayout.Footer>
        </PageLayout>
      </ThemeProvider>
    )

    expect(getByText('Pane')).toBeVisible()
  })
})
