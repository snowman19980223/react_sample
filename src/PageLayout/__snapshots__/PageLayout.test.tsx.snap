// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout renders condensed layout 1`] = `
.c1 {
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c5 {
  width: 100%;
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
}

.c9 {
  width: 100%;
}

.c0 {
  padding: 16px;
}

.c2 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  width: 100%;
  margin-bottom: 16px;
}

.c3 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-top: 16px;
}

.c6 {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-left: 0;
  margin-right: 0;
  margin-top: 16px;
}

.c7 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-bottom: 16px;
}

.c8 {
  height: 100%;
  display: none;
  margin-right: 16px;
}

.c10 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  width: 100%;
  margin-top: 16px;
}

@media screen and (min-width:768px) {
  .c9 {
    width: 256px;
  }
}

@media screen and (min-width:1012px) {
  .c9 {
    width: 296px;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    margin-left: 0 !important;
    margin-right: 0 !important;
    display: none;
  }
}

@media screen and (min-width:768px) {
  .c6 {
    width: auto;
    margin-left: 16px;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-order: 3;
    -ms-flex-order: 3;
    order: 3;
  }
}

@media screen and (min-width:768px) {
  .c7 {
    margin-left: 0 !important;
    margin-right: 0 !important;
    display: none;
  }
}

@media screen and (min-width:768px) {
  .c8 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        Header
        <div
          class="c3"
        />
      </header>
      <main
        class="c4"
      >
        <div
          class="c5"
        >
          Content
        </div>
      </main>
      <aside
        class="c6"
      >
        <div
          class="c7"
        />
        <div
          class="c8"
        />
        <div
          class="c9"
        >
          Pane
        </div>
      </aside>
      <footer
        class="c10"
      >
        <div
          class="c7"
        />
        Footer
      </footer>
    </div>
  </div>
</div>
`;

exports[`PageLayout renders default layout 1`] = `
.c0 {
  padding: 16px;
}

.c1 {
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  width: 100%;
  margin-bottom: 16px;
}

.c3 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-top: 16px;
}

.c4 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c5 {
  width: 100%;
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
}

.c6 {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-left: 0;
  margin-right: 0;
  margin-top: 16px;
}

.c7 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-bottom: 16px;
}

.c8 {
  height: 100%;
  display: none;
  margin-right: 16px;
}

.c9 {
  width: 100%;
}

.c10 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  width: 100%;
  margin-top: 16px;
}

@media screen and (min-width:1012px) {
  .c0 {
    padding: 24px;
  }
}

@media screen and (min-width:1012px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    margin-left: 0 !important;
    margin-right: 0 !important;
    display: none;
  }
}

@media screen and (min-width:1012px) {
  .c3 {
    margin-left: -24px;
    margin-right: -24px;
    margin-top: 24px;
  }
}

@media screen and (min-width:768px) {
  .c6 {
    width: auto;
    margin-left: 16px;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-order: 3;
    -ms-flex-order: 3;
    order: 3;
  }
}

@media screen and (min-width:1012px) {
  .c6 {
    margin-top: 24px;
  }
}

@media screen and (min-width:768px) {
  .c7 {
    margin-left: 0 !important;
    margin-right: 0 !important;
    display: none;
  }
}

@media screen and (min-width:1012px) {
  .c7 {
    margin-left: -24px;
    margin-right: -24px;
    margin-bottom: 24px;
  }
}

@media screen and (min-width:768px) {
  .c8 {
    display: none;
  }
}

@media screen and (min-width:1012px) {
  .c8 {
    margin-right: 24px;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    width: 256px;
  }
}

@media screen and (min-width:1012px) {
  .c9 {
    width: 296px;
  }
}

@media screen and (min-width:1012px) {
  .c10 {
    margin-top: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        Header
        <div
          class="c3"
        />
      </header>
      <main
        class="c4"
      >
        <div
          class="c5"
        >
          Content
        </div>
      </main>
      <aside
        class="c6"
      >
        <div
          class="c7"
        />
        <div
          class="c8"
        />
        <div
          class="c9"
        >
          Pane
        </div>
      </aside>
      <footer
        class="c10"
      >
        <div
          class="c7"
        />
        Footer
      </footer>
    </div>
  </div>
</div>
`;

exports[`PageLayout renders pane in different position when narrow 1`] = `
.c0 {
  padding: 16px;
}

.c1 {
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  width: 100%;
  margin-bottom: 16px;
}

.c3 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-top: 16px;
}

.c4 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c5 {
  width: 100%;
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
}

.c10 {
  margin-left: -16px;
  margin-right: -16px;
  display: none;
  margin-bottom: 16px;
}

.c7 {
  height: 100%;
  display: none;
  margin-right: 16px;
}

.c8 {
  width: 100%;
}

.c9 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  width: 100%;
  margin-top: 16px;
}

.c6 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  width: 100%;
  margin-left: 0;
  margin-right: 0;
  margin-bottom: 16px;
}

@media screen and (min-width:1012px) {
  .c0 {
    padding: 24px;
  }
}

@media screen and (min-width:1012px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    margin-left: 0 !important;
    margin-right: 0 !important;
    display: none;
  }
}

@media screen and (min-width:1012px) {
  .c3 {
    margin-left: -24px;
    margin-right: -24px;
    margin-top: 24px;
  }
}

@media screen and (min-width:768px) {
  .c10 {
    margin-left: 0 !important;
    margin-right: 0 !important;
    display: none;
  }
}

@media screen and (min-width:1012px) {
  .c10 {
    margin-left: -24px;
    margin-right: -24px;
    margin-bottom: 24px;
  }
}

@media screen and (min-width:768px) {
  .c7 {
    display: none;
  }
}

@media screen and (min-width:1012px) {
  .c7 {
    margin-right: 24px;
  }
}

@media screen and (min-width:768px) {
  .c8 {
    width: 256px;
  }
}

@media screen and (min-width:1012px) {
  .c8 {
    width: 296px;
  }
}

@media screen and (min-width:1012px) {
  .c9 {
    margin-top: 24px;
  }
}

@media screen and (min-width:768px) {
  .c6 {
    width: auto;
    margin-left: 16px;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-order: 3;
    -ms-flex-order: 3;
    order: 3;
  }
}

@media screen and (min-width:1012px) {
  .c6 {
    margin-bottom: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        Header
        <div
          class="c3"
        />
      </header>
      <main
        class="c4"
      >
        <div
          class="c5"
        >
          Content
        </div>
      </main>
      <aside
        class="c6"
      >
        <div
          class="c3"
        />
        <div
          class="c7"
        />
        <div
          class="c8"
        >
          Pane
        </div>
      </aside>
      <footer
        class="c9"
      >
        <div
          class="c10"
        />
        Footer
      </footer>
    </div>
  </div>
</div>
`;

exports[`PageLayout renders with dividers 1`] = `
.c0 {
  padding: 16px;
}

.c1 {
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c2 {
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
  width: 100%;
  margin-bottom: 16px;
}

.c4 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

.c5 {
  width: 100%;
  max-width: 100%;
  margin-left: auto;
  margin-right: auto;
}

.c9 {
  width: 100%;
}

.c10 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  width: 100%;
  margin-top: 16px;
}

.c3 {
  margin-left: -16px;
  margin-right: -16px;
  display: block;
  height: 8px;
  background-color: #f6f8fa;
  box-shadow: inset 0 -1px 0 0 #d0d7de,inset 0 1px 0 0 #d0d7de;
  margin-top: 16px;
}

.c6 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  width: 100%;
  margin-left: 0;
  margin-right: 0;
  margin-bottom: 16px;
}

.c7 {
  margin-left: -16px;
  margin-right: -16px;
  display: block;
  height: 8px;
  background-color: #f6f8fa;
  box-shadow: inset 0 -1px 0 0 #d0d7de,inset 0 1px 0 0 #d0d7de;
  margin-top: 16px;
}

.c8 {
  height: 100%;
  display: none;
  margin-left: 16px;
}

.c11 {
  margin-left: -16px;
  margin-right: -16px;
  display: block;
  height: 1px;
  background-color: #d0d7de;
  margin-bottom: 16px;
}

@media screen and (min-width:1012px) {
  .c0 {
    padding: 24px;
  }
}

@media screen and (min-width:1012px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:768px) {
  .c9 {
    width: 256px;
  }
}

@media screen and (min-width:1012px) {
  .c9 {
    width: 296px;
  }
}

@media screen and (min-width:1012px) {
  .c10 {
    margin-top: 24px;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    margin-left: 0 !important;
    margin-right: 0 !important;
    display: block;
    height: 1px;
    background-color: #d0d7de;
  }
}

@media screen and (min-width:1012px) {
  .c3 {
    margin-left: -24px;
    margin-right: -24px;
    margin-top: 24px;
  }
}

@media screen and (min-width:768px) {
  .c6 {
    width: auto;
    margin-right: 16px;
    margin-top: 0 !important;
    margin-bottom: 0 !important;
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }
}

@media screen and (min-width:1012px) {
  .c6 {
    margin-bottom: 24px;
  }
}

@media screen and (min-width:768px) {
  .c7 {
    margin-left: 0 !important;
    margin-right: 0 !important;
    display: none;
  }
}

@media screen and (min-width:1012px) {
  .c7 {
    margin-left: -24px;
    margin-right: -24px;
    margin-top: 24px;
  }
}

@media screen and (min-width:768px) {
  .c8 {
    display: block;
    width: 1px;
    background-color: #d0d7de;
  }
}

@media screen and (min-width:1012px) {
  .c8 {
    margin-left: 24px;
  }
}

@media screen and (min-width:768px) {
  .c11 {
    margin-left: 0 !important;
    margin-right: 0 !important;
    display: none;
  }
}

@media screen and (min-width:1012px) {
  .c11 {
    margin-left: -24px;
    margin-right: -24px;
    margin-bottom: 24px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <header
        class="c2"
      >
        Header
        <div
          class="c3"
        />
      </header>
      <main
        class="c4"
      >
        <div
          class="c5"
        >
          Content
        </div>
      </main>
      <aside
        class="c6"
      >
        <div
          class="c7"
        />
        <div
          class="c8"
        />
        <div
          class="c9"
        >
          Pane
        </div>
      </aside>
      <footer
        class="c10"
      >
        <div
          class="c11"
        />
        Footer
      </footer>
    </div>
  </div>
</div>
`;
